src/main/java/org/takes/misc/Opt.java

```diff
@@ -66,9 +116,14 @@
         /**
          * Ctor.
          * @param orgn Origin
+         * @todo #609:30min null value should not be placed in Opt.Single
+         *  container. TkMethodsTest and TkRetryTest detected problems with
+         *  passing null value to Opt.Single constructor. We should fix these
+         *  problems and re-enable ignored tests.
          */
         public Single(final T orgn) {
-            this.origin = orgn;
+            this.origin =
+                Objects.requireNonNull(orgn, "orgn could not be null");
```

Comment:
@Happy-Neko here I'm wondering, is this class used from outside (initialized by client)? If not then there is no need to check for nulls here, just make sure we don't provide a null to it.